plugins/processors/ec2tagger/ec2tagger.go

```diff
@@ -418,44 +366,47 @@ func (t *Tagger) refreshLoopToUpdateTagsAndVolumes() {
 	}
 }
 
-// updateVolumes calls EC2 describe volume
-func (t *Tagger) updateVolumes() error {
-	if t.ebsVolume == nil {
-		t.ebsVolume = NewEbsVolume()
+/*
+	Retrieve metadata from IMDSv2,IMDSv1 and use these metadata to:
+	* Extract InstanceID, ImageID, InstanceType to create custom dimension for collected metrics
+	* Extract InstanceID to retrieve Instance's Volume and Tags
+	* Extract Region to create aws session with custom configuration
+	For more information on IMDSv2, please follow this document https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html
+*/
+func (t *Tagger) initialRetrievalOfEC2Metadata() error {
+	for _, tag := range t.EC2MetadataTags {
+		switch tag {
+		case mdKeyInstanceId:
+			t.ec2MetadataLookup.instanceId = true
+		case mdKeyImageId:
+			t.ec2MetadataLookup.imageId = true
+		case mdKeyInstanceType:
+			t.ec2MetadataLookup.instanceType = true
+		default:
+			t.Log.Errorf("ec2tagger: Unsupported EC2 Metadata key: %s.", tag)
+		}
 	}
 
-	input := &ec2.DescribeVolumesInput{
-		Filters: []*ec2.Filter{
-			{
-				Name:   aws.String("attachment.instance-id"),
-				Values: aws.StringSlice([]string{t.instanceId}),
-			},
-		},
+	t.Log.Infof("ec2tagger: Check EC2 Metadata.")
+	doc, err := t.ec2MetadataProvider().GetInstanceIdentityDocument()
+	if err != nil {
+		t.Log.Error("ec2tagger: Timeout when retrieving EC2 Metadata. This plugin must only be used on an EC2 instance.")
+		if context.CurrentContext().RunInContainer() {
+			t.Log.Warn("ec2tagger: Timeout may occurred because hop limit is too small. Please increase hop limit to 2 by following this document https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-options.html#configuring-IMDS-existing-instances.")
```

Comment:
Nit-
```suggestion
			t.Log.Warn("ec2tagger: Timeout may have occurred because hop limit is too small. Please increase hop limit to 2 by following  https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-options.html#configuring-IMDS-existing-instances.")
```